mindmeld/models/auto_models.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 Cisco Systems, Inc. and others.  All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Contains classes that can resolve model types and create/load appropriate models
+"""
+
+import logging
+import os
+from typing import Union
+
+from .helpers import register_model
+from .model import ModelConfig, BaseModel
+from .tagger_models import AutoTaggerModel
+from .text_models import AutoTextModel
+
+logger = logging.getLogger(__name__)
+
+
+class AutoModel:
+    """Auto class that identifies appropriate text/tagger model from text_models.py/tagger_models.py
+    to load based on the inputted configs or from the loaded configs file.
+
+    The .from_config() methods allows to load the appropriate model when passed in with ModelConfig.
+    The .from_path() method uses BaseModel's load method to load a dumped config, which is them used
+    to load appropriate model and return it through a metadata dictionary object.
+
+    """
+    ALLOWED_MODEL_TYPES = ["text", "tagger"]
```

Comment:
these can be enums

Reply:
@vijay120 I'm not sure if enum is required here as i am using this var `ALLOWED_MODEL_TYPES` just to register models and nothing else. Do you still think its required?

Reply:
No its fine.